src/app/util/basic-types.h

```diff
@@ -28,17 +28,24 @@
 // Pull in NodeId
 #include <transport/raw/MessageHeader.h>
 
+// Pull in VendorId
+#include <core/CHIPVendorIdentifiers.hpp>
+
 namespace chip {
-typedef uint8_t EndpointId;
-typedef uint16_t ClusterId;
-typedef uint16_t AttributeId;
+typedef uint32_t ClusterId;
+typedef uint32_t AttributeId;
```

Comment:
> This is a little dubious because we're still using 16-bit attribute ids in the ReadAttributes and ReadAttributesResponse commands , no? I am not seeing that changed in `process-global-message.cpp`.
> 
> I'm actually a bit surprised this compiled, since we are passing `attrId` values to `emberAfPutInt16uInResp`. I guess we are not using `-Wconversion` in any of the places that build the process-global-message code? I'll look into fixing that....

IIRC we used to build with `-Wconversion` into various places, such as chip-tool, the build system shift to use the gn template `chip_data_model.gni` may have got rid of `src/app` beeing under the `-Wconversion` radar.

> 
> Given our lack of test/build coverage there, I'm really worried about making these size changes. It might be OK, but it might also make turning on `-Wconversion` pretty hellish....
> 
> At the very least, I would suggest leaving `AttributeId` specifically as 16-bit for now and then making that change in a separate PR after we turn on `-Wconversion` for most of this code.

I locally rebased my PR on top of #7902 and fixed all the places where `-WConversion` was complaining as well as some places where it was not complaining because of hand written cast. 
I did run `./scripts/tests/test_suites.sh`, did some tests for for discover attributes and attributes reporting after all those modifications, so I think that will be OK to keep `AttributeId` with those modifications.